pkg/tasks/tasks.go

```diff
@@ -112,7 +162,18 @@ type TaskSpec struct {
 	Task Task
 }
 
+type Validator interface {
+	Validate(ctx context.Context) *client.ValidationError
```

Comment:
It feels strange to return a struct from the `client` package.

Reply:
I think we should must return `TaskValidationError` instead. Good point! 🙇

Reply:
We can't return `TaskValidationError` which needs to know the `name` of the Task. Hence I am aliasing client.ValidationError to ValidationError